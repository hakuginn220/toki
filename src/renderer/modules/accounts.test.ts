import reducer, { ActionTypes, initialState } from './accounts'

let count = 0

const createUser = () => {
  count = count + 1

  return {
    access_token: `token code for No.${count}`,
    access_token_secret: `secret code for No.${count}`
  }
}

test('ADD_ACCOUNT', () => {
  expect(
    reducer(initialState, {
      payload: { user: createUser() },
      type: ActionTypes.ADD_ACCOUNT
    })
  ).toMatchSnapshot()
})

test('REMOVE_ACCOUNT', () => {
  let state

  for (let i = 0; i < 3; i++) {
    state = reducer(state, {
      payload: { user: createUser() },
      type: ActionTypes.ADD_ACCOUNT
    })
  }

  expect(
    reducer(state, {
      payload: { index: 1 },
      type: ActionTypes.REMOVE_ACCOUNT
    })
  ).toMatchSnapshot()
})

test('CHANGE_VERIFIER', () => {
  expect(
    reducer(initialState, {
      payload: { verifier: 'authentication code' },
      type: ActionTypes.CHANGE_VERIFIER
    })
  ).toMatchSnapshot()
})
