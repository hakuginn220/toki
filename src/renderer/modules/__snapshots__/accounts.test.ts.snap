// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADD_ACCOUNT 1`] = `
Object {
  "users": Array [
    Object {
      "access_token": "token code for No.1",
      "access_token_secret": "secret code for No.1",
    },
  ],
  "verifier": "",
}
`;

exports[`CHANGE_VERIFIER 1`] = `
Object {
  "users": Array [],
  "verifier": "authentication code",
}
`;

exports[`OPEN_AUTHORIZE 1`] = `
Object {
  "users": Array [],
  "verifier": "",
}
`;

exports[`REMOVE_ACCOUNT 1`] = `
Object {
  "users": Array [
    Object {
      "access_token": "token code for No.2",
      "access_token_secret": "secret code for No.2",
    },
    Object {
      "access_token": "token code for No.4",
      "access_token_secret": "secret code for No.4",
    },
  ],
  "verifier": "",
}
`;
